hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/file/FileTableFunction.java

```diff
@@ -83,6 +95,52 @@ private static String randomName() {
         return SCHEMA_NAME_FILES + "_" + UuidUtil.newUnsecureUuidString().replace('-', '_');
     }
 
+    @Override
+    public List<Permission> permissions(SqlCall call) {
+        // Try find operand by name first, most common option.
+        SqlNode astPath = findOperandByName(OPTION_PATH, call);
+        // If not successful -> try to find by position: SELECT * FROM TABLE(fn('path', 'glob'))
+        if (astPath == null) {
+            astPath = findOperandByPosition(0, call);
+        }
+        if (astPath == null) {
+            throw QueryException.error("Can't determine the file path, query is denied in secure environment");
+        }
+        if (astPath instanceof SqlLiteral) {
+            String path = astPath.toSqlString(PostgresqlSqlDialect.DEFAULT).getSql();
+            // Strip quotes
+            if (path.charAt(0) == '\'' && path.charAt(path.length() - 1) == '\'') {
+                path = path.substring(1, path.length() - 1);
+            }
+            return Collections.singletonList(ConnectorPermission.file(path, ActionConstants.ACTION_READ));
+        } else if (astPath instanceof SqlDynamicParam) {
+            // Note: it is pre-validation phase, we can't extract the path value from dynamic param yet.
+            //  Better to have more strict permissions check here.
+            return Collections.singletonList(ConnectorPermission.file("*", ActionConstants.ACTION_READ));
```

Comment:
Do I understand that if I do:
```
hz.getSql().execute("SELECT * FROM TABLE(\n"
                                + " JSON_FLAT_FILE(path => ?, glob => ?)\n)",
                        "/data/", "input.json")
```
I need file permissions for "*" - all files, but if I do 
```
hz.getSql().execute("SELECT * FROM TABLE(\n"
                                + " JSON_FLAT_FILE(path => 'data', glob => 'input.json')\n)")
```
It is sufficient to have permission for `/data` only.

Reply:
@frant-hartm yes, but there is a note above: 
```
 // Note: it is pre-validation phase, we can't extract the path value from dynamic param yet.
 //  Better to have more strict permissions check here.
```